server/src/com/thoughtworks/go/server/web/BackupFilter.java

```diff
@@ -0,0 +1,173 @@
+            return;
+        }
+        if (backupService.isBackingUp()) {
+            if (isAPIUrl(url) && !isMessagesJson(url)) {
+                generateAPIResponse(request, response);
+            } else {
+                generateHTMLResponse(response);
+            }
+        } else {
+            chain.doFilter(request, response);
+        }
+
+    }
+
+    private void generateHTMLResponse(ServletResponse response) {
+        String path = "backup_in_progress.html";
+        InputStream resourceAsStream = getClass().getClassLoader().getResourceAsStream(path);
+        response.setContentType("text/html");
+        StringWriter writer = new StringWriter();
+        try {
+            IOUtils.copy(resourceAsStream, writer);
+            String content = writer.toString();
+            content = replaceStringLiterals(content);
+            response.getWriter().print(content);
+        } catch (FileNotFoundException e) {
+            LOGGER.error(String.format("File not found: %s", path));
+        } catch (IOException e) {
+            LOGGER.error(String.format("General IOException: %s", e.getMessage()));
+        }
+    }
+
+    String replaceStringLiterals(String content) {
+        content = content.replaceAll("%backup_initiated_by%", HtmlUtils.htmlEscape(backupService.backupRunningSinceISO8601()));
+        content = content.replaceAll("%backup_started_by%", HtmlUtils.htmlEscape(backupService.backupStartedBy()));
+        return content;
+    }
+
+    private void generateAPIResponse(ServletRequest request, ServletResponse response) {
+
+        try {
+            HttpServletRequest httpRequest = (HttpServletRequest) request;
+
+            String message = "Server is under maintenance mode, please try later.";
+
+            if (requestIsOfType(JSON, httpRequest)) {
+                response.setContentType("application/json");
+                JsonObject jsonObject = new JsonObject();
+                jsonObject.addProperty("message", message);
+                response.getWriter().print(jsonObject);
+            } else if (requestIsOfType(XML, httpRequest)) {
+                response.setContentType("application/xml");
+                String xml = String.format("<message> %s </message>", message);
+                response.getWriter().print(xml);
+            } else {
+                generateHTMLResponse(response);
+            }
+
+        } catch (IOException e) {
+            LOGGER.error(String.format("General IOException", e.getMessage()));
+        }
+        ((HttpServletResponse) response).setStatus(503);
+    }
+
+    private boolean requestIsOfType(String type, HttpServletRequest request) {
+        String header = request.getHeader("Accept");
+        String contentType = request.getContentType();
+        String url = request.getRequestURI().toString();
+        return header != null && header.contains(type) || url != null && url.contains(type) || contentType != null && contentType.contains(type);
```

Comment:
did you mean `url.endsWith`? This will return unexpected response for a url that looks like `/go/tab/pipeline/history/build-xml-document`